### ISSUE TITLE
Question on the implementation of the function BlockNonlinearForm::SetEssentialBC 

### ISSUE THREAD

**Issue by AdrainT** [mfem/mfem#3951]
Hi, I am using MFEM's BlockNonlinearForm, and is reading the codes. The codes for BlockNonlinearForm::SetEssentialBC is as follows:

 void BlockNonlinearForm::SetEssentialBC(
    const Array<Array<int> *> &bdr_attr_is_ess, Array<Vector *> &rhs)
 {
    for (int s = 0; s < fes.Size(); ++s)
    {
       ess_tdofs[s]->SetSize(ess_tdofs.Size());
 
       fes[s]->GetEssentialTrueDofs(*bdr_attr_is_ess[s], *ess_tdofs[s]);
 
       if (rhs[s])
       {
          rhs[s]->SetSubVector(*ess_tdofs[s], 0.0);
       }
    }
 }

My question is, what's the meaning of "ess_tdofs[s]->SetSize(ess_tdofs.Size())" ? From my understanding, ess_tdofs.Size() is the number of blocks, while ess_tdofs[s] is the number of essential true dofs in each fespace. They are different. Is my understanding wrong?

My mfem version is 4.6.

**Comment by miaodi** [comment]
I agree with you. Essentially, `ess_tdofs[s]->SetSize(ess_tdofs.Size());`  is redundant, as `MarkerToList` call in `GetEssentialTrueDofs` will modify the size of `*ess_tdofs[s]`.

**Comment by v-dobrev** [comment]
Yes, it looks like the line `ess_tdofs[s]->SetSize(ess_tdofs.Size());` is unnecessary. @AdrainT, do you want to make a PR to fix it?
